kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/ReplicaManager.java

```diff
@@ -111,8 +111,12 @@ public PartitionLog getPartitionLog(TopicPartition topicPartition, String namesp
                         .thenAccept(offset -> addPartitionResponse.accept(topicPartition,
                                 new ProduceResponse.PartitionResponse(Errors.NONE, offset, -1L, -1L)))
                         .exceptionally(ex -> {
+                            Errors errors = Errors.forException(ex.getCause());
+                            if (errors == Errors.UNKNOWN_SERVER_ERROR) {
+                                errors = Errors.KAFKA_STORAGE_ERROR;
+                            }
```

Comment:
Could you explain why do you treat the `UNKNOWN_SERVER_ERROR` as `KAFKA_STORAGE_ERROR` here?